torch/distributed/rendezvous.py

```diff
@@ -50,6 +50,13 @@ def register_rendezvous_handler(scheme, handler):
         )
     _rendezvous_handlers[scheme] = handler
 
+# Query will have format "rank=0&world_size=1" and is
+# converted into {"rank": 0, "world_size": 1}
+def _query_to_dict(query):
+    query_dict: Dict[str, Union[int, str]] = dict(
+        cast(Tuple[str, Union[int, str]], pair.split("=")) for pair in cast(Iterable[str], filter(None, query.split("&")))
```

Comment:
curious, any reason that we need a filter given the condition is `None`?

Reply:
Can this be simplified to the following?

```python
dict(cast(Tuple[str, Union[int, str]], pair.split("=")) for pair in query.split("&"))
```

Reply:
Regarding `Tuple[str, Union[int, str]]`, when will the key be int? For input `"rank=0&world_size=1" `, looks like the output is the following?

```python
{'rank': '0', 'world_size': '1'}
```

Reply:
> curious, any reason that we need a filter given the condition is None?

The None filters out any empty values, such as if the query contained `"rank=0&world_size=1&"` (note the extra &)

> Can this be simplified to the following? 
```python 
dict(cast(Tuple[str, Union[int, str]], pair.split("=")) for pair in query.split("&"))
```

Need the cast to Iterable[str] or else mypy complains `torch/distributed/rendezvous.py:56:38: error: Need type annotation for 'pair'  [var-annotated]`

> Regarding Tuple[str, Union[int, str]], when will the key be int? 

Good point, it will not be int. `Tuple[str, str]` is sufficient.